reana_server/status.py

```diff
@@ -0,0 +1,220 @@
+            Workflow.interactive_session.isnot(None)).all()
+        return len(active_interactive_sessions)
+
+    def get_status(self):
+        """Get status summary for interactive sessions."""
+        return {
+            'active': self.get_active(),
+        }
+
+
+class SystemStatus(REANAStatus):
+    """Class to retrieve statistics related to REANA interactive sessions."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise SystemStatus class."""
+        super().__init__(from_=from_, until=until, user=user)
+
+    def uptime(self):
+        """Get component uptime."""
+        cmd = ['uptime', '-p']
+        return self.execute_cmd(cmd)
+
+    def get_status(self):
+        """Get status summary for REANA system."""
+        return {
+            'uptime': self.uptime(),
+        }
+
+
+class StorageStatus(REANAStatus):
+    """Class to retrieve statistics related to REANA storage."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise StorageStatus class."""
+        super().__init__(from_=from_, until=until, user=user)
+
+    def _get_path(self):
+        """Retrieve the path to calculate status from."""
+        path = None
+        if self.user:
+            path = self.user.workspace_path
+        else:
+            path = SHARED_VOLUME_PATH + '/users'
+
+        return path
+
+    def users_directory_size(self):
+        """Get disk usage for users directory."""
+        depth = 0
+        cmd = ['du', '-h', f'--max-depth={depth}', self._get_path()]
+        output = self.execute_cmd(cmd)
+        size = output.split()[0]
+        return size
+
+    def shared_volume_health(self):
+        """REANA shared volume health."""
+        cmd = ['df', '-h', SHARED_VOLUME_PATH]
+        output = self.execute_cmd(cmd).splitlines()
+        headers = output[0].split()
+        values = output[1].split()
+        used_index = headers.index('Used')
+        available_index = headers.index('Avail')
+        use_percentage_index = headers.index('Use%')
+
+        return f'{values[used_index]}/{values[available_index]} ' \
+               f'({values[use_percentage_index]})'
+
+    def get_status(self):
+        """Get status summary for REANA storage."""
+        return {
+            'user_directory_size': self.users_directory_size(),
+            'shared_volume_health': self.shared_volume_health(),
+        }
+
+
+class UsersStatus(REANAStatus):
+    """Class to retrieve statistics related to REANA users."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise UsersStatus class."""
+        super().__init__(from_=from_, until=until, user=user)
+
+    def active_web_users(self):
+        """Number of active web users.
+
+        Depends on how long does a session last.
+        """
+        number = (Session.query(SessionActivity)
+                  .filter(
+                      SessionActivity.updated.between(self.from_, self.until))
```

Comment:
This is the only class using `from_` and `until` I assume is WIP.

Reply:
I think we are not in the moment to tackle as almost no feature can support it. Probably we should leave the code as is, removing the from_ and until from the CLI (completes https://github.com/reanahub/reana-server/pull/240#discussion_r417354844)